frontend/components/suggestion-and-search.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import debounce from 'lodash.debounce';
+import { Badge } from './badge-examples';
+
+function toXhrSend(data) {
+  let str = '', start = true;
+  let jsondata = '';
+  for (const key in data) {
+    if (typeof (jsondata = JSON.stringify(data[key])) === 'string') {
+      str += start ? '' : '&';
+      if (typeof data[key] === 'string') {
+        str += encodeURIComponent(key) + '=' + encodeURIComponent(data[key]);
+      } else {
+        str += encodeURIComponent(key) + '=' + encodeURIComponent(jsondata);
```

Comment:
This is doing exactly the same encoding as your `makeBadgeUri` method in `DynamicBadgeMaker`. Let's pull this out into a common utility file.